### PR TITLE
Update build configurations

### PR DESCRIPTION
**Pull request by keithc-ca**
* update boot jdk for jdknext to Java 16
* remove information relating to jdk15

### CODE REVIEW

File: buildenv/jenkins/variables/defaults.yml [hunk 1]
```diff
@@ -170,9 +165,8 @@ boot_jdk_default:
     version:
       8: '8'
       11: '11'
-      15: '15'
       16: '15'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml:168]: This can change to 16 as well, then we don't need to have 15 on the build machines any more.

**Comment by pshipton** [comment]
jenkins compile all jdknext

File: buildenv/jenkins/variables/defaults.yml (hunk 1 above)
Comment (keithc-ca) [reply buildenv/jenkins/variables/defaults.yml:168]: I suppose the risk is low that we have (or mask) build problems using the newer bootjdk; I'll update that.

Comment (keithc-ca) [reply buildenv/jenkins/variables/defaults.yml]: Done.

**Comment by keithc-ca** [comment]
Jenkins compile all jdk16,jdknext

**Comment by pshipton** [comment]
Close enough, there are infra problems this morning.

**Review by pshipton — APPROVED** [review]

**Pull request merged**